### PR TITLE
SCTP support for pod2service scenarios in uperf workloads

### PR DESCRIPTION
**Pull request by josecastillolema**
### Description
Besides `tcp` and `udp`, we have added `sctp` support for uperf workloads service scenarios.

### CODE REVIEW

File: roles/uperf/templates/service_metallb.yml.j2 [hunk 1]
```diff
@@ -82,6 +82,47 @@ items:
...
+{% endif %}
+      externalTrafficPolicy: '{{ workload_args.metallb.service_etp | default("Cluster") }}'
+      type: LoadBalancer
+      ports:
+      - name: uperf
+        port: 30000
+        targetPort: 30000
+        protocol: SCTP
+{% for num in range(30001,30012,1) %}
+      - name: uperf-control-tcp-{{num}}
```
Comment (jtaleric) [inline roles/uperf/templates/service_metallb.yml.j2:120]: You mention `uperf-control-tcp` here, but have the protocol as SCTP? 

Is the proto sctp or tcp here? This is for the uperf control - not sure if it should be sctp or tcp... I haven't messed with sctp, so flying blind :(

**Comment by rsevilla87** [comment]
@josecastillolema seems like with the latest changes this one needs some rebase

File: roles/uperf/templates/service_metallb.yml.j2 (hunk 1 above)
Comment (josecastillolema) [reply roles/uperf/templates/service_metallb.yml.j2:120]: We are following the same naming convection for SCTP like the one used for TCP and UDP: https://github.com/cloud-bulldozer/benchmark-operator/blob/84d7f13ba2b0de41ad359ddb1c8b529f0a3ba1e2/roles/uperf/templates/service_metallb.yml.j2#L75-L83

**Comment by josecastillolema** [comment]
I understand [bats](https://github.com/bats-core/bats-detik) is being used to run the k8s e2e tests. Would it be possible (or even worth it) to [enable SCTP](https://docs.openshift.com/container-platform/4.10/networking/using-sctp.html#nw-sctp-enabling_using-sctp) on the e2e cluster to include SCTP tests?
We have to take into consideration that the enablement of SCTP requires a reboot of the host.

**Comment by rsevilla87** [comment]
> I understand [bats](https://github.com/bats-core/bats-detik) is being used to run the k8s e2e tests. Would it be possible (or even worth it) to [enable SCTP](https://docs.openshift.com/container-platform/4.10/networking/using-sctp.html#nw-sctp-enabling_using-sctp) on the e2e cluster to include SCTP tests? We have to take into consideration that the enablement of SCTP requires a reboot of the host.

Yeah, it would be nice to have an e2e test for this scenario, uperf tests are specified at https://github.com/cloud-bulldozer/benchmark-operator/blob/master/e2e/013-uperf.bats, it will be required to add your testcase there.

In the meantime, Im gonna enable sctp in the CI cluster, I'll let you know when ready.

**Comment by rsevilla87** [comment]
Hey @josecastillolema, just finished enabling SCP in the CI cluster, feel free to add the test case now.

**Comment by josecastillolema** [comment]
Thanks @rsevilla87 !
About the tests, I was thinking in including them here: 
https://github.com/cloud-bulldozer/benchmark-operator/blob/master/tests/test_crs/valid_uperf_serviceip.yaml#L34, is that ok?

**Comment by rsevilla87** [comment]
> Thanks @rsevilla87 ! About the tests, I was thinking in including them here: https://github.com/cloud-bulldozer/benchmark-operator/blob/master/tests/test_crs/valid_uperf_serviceip.yaml#L34, is that ok?

The tests directory is deprecated and just used by benchmark-wrapper. If you want to add a new test case you have to update https://github.com/cloud-bulldozer/benchmark-operator/blob/master/e2e/013-uperf.bats and add your new CR file to https://github.com/cloud-bulldozer/benchmark-operator/tree/master/e2e/uperf

**Review by rsevilla87 — APPROVED** [review]
Seems like the e2e test passed w/o issues, LGTM!

BTW, I guess that this PR is also assing sctp support to the pod2pod scenario, doesn't it?

**Comment by josecastillolema** [comment]
> Seems like the e2e test passed w/o issues, LGTM!
> 
> BTW, I guess that this PR is also adding sctp support to the pod2pod scenario, doesn't it?

We are about to find out

**Pull request merged**